### PR TITLE
Migrate github source off reconciler.Base

### LINKED ISSUE #1076: migrate github source off reconciler.Base
We are attempting to remove `reconciler.Base`. Migration needs to happen for this component.

See the [examples](https://github.com/knative/eventing/pulls?q=is%3Apr+reconciler.Base+is%3Aclosed).

### PR DESCRIPTION
**Pull request by tom24d**
Fixes #1076

### CODE REVIEW

File: github/pkg/reconciler/githubsource.go [hunk 1]
```diff
@@ -19,13 +19,15 @@ package github
 import (
 	"context"
 	"fmt"
```
Comment (mattmoor) [inline github/pkg/reconciler/githubsource.go]: Format Go code:
```suggestion
	"fmt"
	"strings"

```

File: github/pkg/reconciler/githubsource.go [hunk 2]
```diff
@@ -19,13 +19,15 @@ package github
...
+	"knative.dev/pkg/controller"
 	"strings"
```
Comment (mattmoor) [inline github/pkg/reconciler/githubsource.go]: Format Go code:
```suggestion

```

File: github/pkg/reconciler/githubsource.go [hunk 3]
```diff
@@ -37,14 +38,15 @@ import (
 	ghreconciler "knative.dev/eventing-contrib/github/pkg/client/injection/reconciler/sources/v1alpha1/githubsource"
 	"knative.dev/eventing-contrib/github/pkg/reconciler/resources"
 
-	"knative.dev/eventing/pkg/reconciler"
 	duckv1 "knative.dev/pkg/apis/duck/v1"
 
 	//knative.dev/pkg imports
 	"knative.dev/pkg/apis"
```
Comment (mattmoor) [inline github/pkg/reconciler/githubsource.go]: Format Go code:
```suggestion
	"knative.dev/pkg/apis"
	"knative.dev/pkg/controller"
```

File: github/pkg/reconciler/githubsource.go [hunk 4]
```diff
@@ -37,14 +38,15 @@ import (
...
 	"knative.dev/pkg/logging"
 	pkgreconciler "knative.dev/pkg/reconciler"
 	"knative.dev/pkg/resolver"
+	"knative.dev/pkg/controller"
+
```
Comment (mattmoor) [inline github/pkg/reconciler/githubsource.go]: Format Go code:
```suggestion

```

File: github/pkg/reconciler/githubsource.go [hunk 5]
```diff
@@ -37,14 +38,15 @@ import (
...
-	"knative.dev/eventing/pkg/reconciler"
 	duckv1 "knative.dev/pkg/apis/duck/v1"
 
 	//knative.dev/pkg imports
 	"knative.dev/pkg/apis"
+	"knative.dev/pkg/controller"
 	"knative.dev/pkg/logging"
 	pkgreconciler "knative.dev/pkg/reconciler"
 	"knative.dev/pkg/resolver"
+
```
Comment (mattmoor) [inline github/pkg/reconciler/githubsource.go]: Format Go code:
```suggestion

```

**Comment by n3wscott** [comment]
/ok-to-test

oooh lookie lookie

**Comment by tom24d** [comment]
Mmm, I have no idea what's happening because I don't do anything with Kafka,,.

**Comment by tom24d** [comment]
Maybe is it related to [knative.dev/eventing/pull/2854](https://github.com/knative/eventing/pull/2854)?
If it is, I assume that all e2e test will be failed.
What is next step?

**Comment by tom24d** [comment]
Oh sorry seems like I do not have to merge that :(

**Pull request merged**